grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -849,7 +844,7 @@ public boolean fixDuplicateResources(String index, Set<ConceptId> resourceVertex
 
             //Restore the index
             String newIndex = mainResource.getIndex();
-            mainResource.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
+            mainResource.vertex().element().property(Schema.ConceptProperty.INDEX.name(), newIndex);
```

Comment:
This should use the existing `property` setter on `VertexElement`. There are probably other cases like this that I missed as well.